crest/Assets/Crest/Crest/Shaders/OceanInputs/Resources/AnimWavesGenerate.shader

```diff
@@ -0,0 +1,170 @@
+﻿// Crest Ocean System
```

Comment:
Does this replace AnimWavesGerstner shaders?

Reply:
it does not.. at the moment..

AnimWavesGerstner is used by Gerstner stuff. Gerstner stuff does not support painted input. it could be made to but I'm not sure if it makes sense, at least from current knowledge. there might be a good use case for it though.

AnimWavesGerstnerGeometry is unfortunately used by both FFT and Gerstner for the spline case (or for user-provided geometry, in theory at least). This is a case of bad naming. We could rename to AnimWavesGenerateFromGeometry potentially?

Reply:
> AnimWavesGerstner is used by Gerstner stuff. Gerstner stuff does not support painted input. it could be made to but I'm not sure if it makes sense, at least from current knowledge. there might be a good use case for it though.

It looks like they can use the same shader as painted is keyword driven. Maybe the changes could be added to Gerstner shader and then we can do a rename in separate PR? That would also improve the diff. LMKWYT

> AnimWavesGerstnerGeometry is unfortunately used by both FFT and Gerstner for the spline case (or for user-provided geometry, in theory at least). This is a case of bad naming. We could rename to AnimWavesGenerateFromGeometry potentially?

Sounds good. I've added it to Trello.

Reply:
> Maybe the changes could be added to Gerstner shader and then we can do a rename in separate PR?

I didn't understand, could you speak more to this?

Reply:
> I didn't understand, could you speak more to this?

Sure. Add painting changes to AnimWavesGerstner.shader instead of creating a new file and then rename it. It should keep git history then. Separate PR might be nice as need to rename AnimWavesGerstner.shader and AnimWavesGerstnerGeometry.shader.

Reply:
ohh got it. alright will do